ngi_visualizations/fpkm_scatter/fpkm_scatter.py

```diff
@@ -0,0 +1,253 @@
+    
+    # We have a summary file
+    else:
+        # Parse the input files
+        # returns counts[sample_name][gene_id] = fpkm_count
+        condition_1 = load_summary_fpkm_counts(input_1)
+        condition_2 = load_summary_fpkm_counts(input_2)
+        
+        # Condition basenames
+        cond_1_basename = os.path.splitext(os.path.basename(input_1))[0]
+        cond_2_basename = os.path.splitext(os.path.basename(input_2))[0]
+        
+        # Find project names from sample ID (P1234_*)
+        pname_re = re.compile('^P\d+_')
+        cond_1_proj = pname_re.search(condition_1.keys()[0]).group(0)
+        cond_2_proj = pname_re.search(condition_2.keys()[0]).group(0)
+        
+        # Go through each sample pair
+        for cond_1_sample in condition_1.keys():
+            try:
+                cond_2_sample = cond_1_sample.replace(cond_1_proj, cond_2_proj)
+                condition_2[cond_2_sample]
+                outfile = "{}_{}-{}_{}".format(cond_1_basename, cond_1_sample, cond_2_basename, cond_2_sample)
+                
+                plot_filenames = plot_fpkm_scatter(condition_1[cond_1_sample], condition_2[cond_2_sample], cond_1_sample, cond_2_sample, output_fn=outfile, linear=linear)
+            except KeyError:
+                logging.warning("Warning: Sample {} not found in {}".format(sample, cond_2_basename))
+    
+    
+def load_fpkm_counts (file):
+    """
+    Reads FPKM file and returns a dict of counts.
+    Input: Path to FPKM file
+    Output: Dictionary. Keys are transcript names, values are counts.
+    """
+    
+    counts = {}
+    logging.info("Processing {}".format(file))
+    if os.path.isfile(file):
+        try:
+            with open(file, 'r') as fh:
+                # Skip the header
+                next(fh)
+                # Iterate through the file
+                for line in fh:
+                    line = line.strip()
+                    cols = line.split("\t")
+                    gene_id = cols[0]
+                    FPKM = cols[9]
+                    counts[gene_id] = FPKM
+        except IOError as e:
+            logging.error("Error loading FPKM file: {}".format(file))
+            raise IOError(e)
+    else:
+        raise IOError("Couldn't find FPKM file: {}".format(file))
+    
+    return counts
+
+
+def load_summary_fpkm_counts (file):
+    """
+    Reads FPKM summary file and returns counts.
+    Input: Path to summary FPKM file
+    Output: Nested Defaultdict. First keys are sample names,
+            second keys are gene names, values are counts.
+            counts[sample_name][gene_id] = fpkm_count
+    """
+    
+    counts = {}
+    
+    # ENSEMBL_ID    Gene_ID    Sample_1    Sample_2    Sample_3    Sample_4
+    # ENSG00000000003    TSPAN6    0.1234    0.1234    0.1234    0.1234
+    
+    # Read the counts file
+    if os.path.isfile(file):
+        try:
```

Comment:
I don't get this. Why test whether file is a file, and then also do a try? And why are the error messages different? Why not just use try only?